Libraries/LibWeb/HTML/HTMLSelectElement.cpp

```diff
@@ -691,4 +697,29 @@ bool HTMLSelectElement::is_focusable() const
     return enabled();
 }
 
+// https://html.spec.whatwg.org/#placeholder-label-option
+HTMLOptionElement* HTMLSelectElement::placeholder_label_option() const
+{
+    // If a select element has a required attribute specified, does not have a multiple attribute specified, and has a display size of 1;
+    if (has_attribute(HTML::AttributeNames::required) && !has_attribute(HTML::AttributeNames::multiple) && display_size() == 1) {
+        // and if the value of the first option element in the
+        // select element's list of options (if any) is the empty string, and that option element's parent node is the select element (and not an optgroup element), then that option is the
+        // select element's placeholder label option.
+        auto first_option_element = list_of_options()[0];
+        if (first_option_element->value().is_empty() && first_option_element->parent() == this)
+            return first_option_element;
+    }
+    return {};
+}
+
+// https://html.spec.whatwg.org/#the-select-element:suffering-from-being-missing
+bool HTMLSelectElement::suffering_from_being_missing() const
+{
+    // If the element has its required attribute specified, and either none of the option elements in the select element's list of options have their selectedness
+    // set to true, or the only option element in the select element's list of options with its selectedness set to true is the placeholder label option, then the element is suffering from being
+    // missing.
+    // FIXME: Add the stuff about the placeholder label option
```

Comment:
Looks like this FIXME is already done?

Reply:
Oh yea, good point.
I put that in a bit later, forgot to remove the fixme.